state/cleanup.go

```diff
@@ -384,17 +384,24 @@ func (st *State) cleanupMachinesForDyingModel(cleanupArgs []bson.Raw) (err error
 		return errors.Trace(err)
 	}
 	force := args.Force != nil && *args.Force
+
+	// There are some cases below, where we should continue on destroying
+	// machines if a failure is encountered with one machine.
+	//   1) If destroy machine fails due to existing containers on it.
+	//   2) If force is in use.
+	var continueError bool
 	for _, m := range machines {
 		if m.IsManager() {
 			continue
 		}
-		if _, isContainer := m.ParentId(); isContainer {
-			continue
-		}
```

Comment:
This doesn't Iook right to remove this continue. The idea was that we don't explicitly destroy the containers themselves - we destroy their parent machine as that is what is provisioned within the cloud. What's the causal link between this code and a failing model destroy?

Reply:
machine-0: 18:20:35 WARNING juju.state cleanup failed in model c62fd152-47b9-4f26-840c-d36d22438ae1 for modelMachines("c62fd152-47b9-4f26-840c-d36d22438ae1"): while destroying machine 0 is: machine 0 is hosting containers "0/lxd/0,0/lxd/1"

Destroy for machines calls advancesLifecycle, which fails if there are any containers on the machine: https://github.com/juju/juju/blob/de13c96e76715720a71d23d65bc6aab57dd8ee06/state/machine.go#L777